### ISSUE TITLE
[bug] Link error on windows with any package installed by Conan

### ISSUE THREAD

**Issue by shelper** [conan-io/conan#6948]
### Environment Details (include every applicable attribute)
  * Operating System+version: windows 10
  * Compiler+version: visual studio Professional 2019 amd64_x86
  * Conan version: 1.24.1
  * Python version: 3.6

### Steps to reproduce (Include if Applicable)
Using the default profile:
```
[settings]
os=Windows
os_build=Windows
arch=x86_64
arch_build=x86_64
compiler=Visual Studio
compiler.version=16
build_type=Release
[options]
[build_requires]
[env]
```
have a very simple project using gtest. 
so i install gtest by : conan install .. -b 
and use cmake to build the project, i got link error:
```
[build] Starting build
[proc] Executing command: "C:\Program Files\CMake\bin\cmake.EXE" --build c:/Users/MPNV38/Develop/vscode_cpp_cmake_example/build --config Debug --target ExampleFunc_test -- /maxcpucount:14
[build] Microsoft (R) Build Engine version 16.5.0+d4cbfca49 for .NET Framework
[build] Copyright (C) Microsoft Corporation. All rights reserved.
[build] 
[build]   ExampleFunc.vcxproj -> C:\Users\MPNV38\Develop\vscode_cpp_cmake_example\build\lib\ExampleFunc.lib
[build] gmock_main.lib(gtest-all.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in ExampleFunc_test.obj [C:\Users\username\Develop\vscode_cpp_cmake_example\build\test\ExampleFunc_test.vcxproj]
[build] gmock_main.lib(gtest-all.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MDd_DynamicDebug' in ExampleFunc_test.obj [C:\Users\username\Develop\vscode_cpp_cmake_example\build\test\ExampleFunc_test.vcxproj]
[build] gmock_main.lib(gtest-all.obj) : warning LNK4099: PDB 'gmock_main.pdb' was not found with 'gmock_main.lib(gtest-all.obj)' or at 'C:\Users\username\Develop\vscode_cpp_cmake_example\build\bin\gmock_main.pdb'; linking object as if no debug info [C:\Users\username\Develop\vscode_cpp_cmake_example\build\test\ExampleFunc_test.vcxproj]
[build] ExampleFunc.lib(ExampleFunc.cpp.obj) : fatal error LNK1112: module machine type 'x86' conflicts with target machine type 'x64' [C:\Users\username\Develop\vscode_cpp_cmake_example\build\test\ExampleFunc_test.vcxproj]
[build] Build finished with exit code 
```

**Comment by memsharded** [comment]
Hi @shelper 

Are you using pure Visual Studio, with the ``visual_studio`` generator?
It seems that the configuration of your Visual is not matching the one you are using to install your dependencies. For example you are installing with the default configuration (Release), but Visual Studio is building with Debug. And is it possible that you are building for x86, while your Conan architecture is ``arch=x86_64``?

**Comment by shelper** [comment]
you are right, i am using debug for building my project but release for installing conan packages. 

This is different from linux or other platforms, in linux it is fine for you to install release version packages and link to them with your debug version project 

I think it would be nice if this can be noted (and probably explained) in conan's documentation. 

I am closing this one, thanks!

**Comment by dragondive** [comment]
I found this post by searching for error message I got when building with Visual Studio and cmake. (I don't use the Visual Studio IDE/editor, only the compiler.) The above comments from @memsharded and @shelper helped me understand and resolve the issue. I'm posting the commands I used so that it may help someone else in future. 

My `conan install` settings were the same (specifically `arch=x86_64` and `build_type=Release`). Hence I invoked `cmake` to match it:

    cmake .. -A x64  -DCMAKE_BUILD_TYPE=Release
    
The type should _also_ be changed in the build call:

    cmake --build . --config Release

Now the build is successful and the executable is generated.
